### DISCUSSION TITLE
Cookiecutter Web Map Template

### DISCUSSION THREAD

**Discussion by PhilipMathieu** [protomaps/PMTiles#367] [Show and tell]
Hey y'all,

I posted previously about some of my work helping integrate geospatial technologies into the data science program at my grad school (Northeastern University, Portland Maine campus). After exploring a lot of different options, I set up a `cookiecutter` template at https://github.com/PhilipMathieu/cookiecutter-bootstrap-maplibre-pmtiles

The template sets up a simple project with the following structure:
```
.
├── LICENSE.md
├── README.md
├── data
│   └── example
│       ├── DATASET.md
│       └── ski_areas_all.geojson
└── docs
    ├── index.html
    └── tiles
        └── ski_areas_all.pmtiles
```
The README explains how to take a new GeoJSON file, generate a pmtiles file, and add it to the map in docs/index.html. I tried to keep the HTML code as minimal as possible while including some of the most important "nice to haves" - namely, basic column layouts, full-width map, and ability to override colors. [You can view the sample project here.](https://github.com/PhilipMathieu/interactive-web-map)

Would love any comments or feedback! I'm hoping to "beta test" this with some of my fellow students later this spring.

**Comment by bdon** [comment]
@PhilipMathieu Looks nice so far!

I would suggest having a large dataset > 5MB or so to clarify the motivation for using PMTiles over GeoJSON, a GeoJSON of <1MB can be loaded directly in the browser without the complexity of that tiling conversion step.

For templates you could also try the new Observable Framework, I made a leaflet/maplibre/pmtiles example here: https://github.com/bdon/observable-framework-maps

**Reply by PhilipMathieu** [reply]
This is awesome - I wasn't aware of the Source Cooperative tileset and will probably change that to the basemap.

I agree that a larger dataset would be better for showing the motivation, but decided not to include it in the template repo just to avoid making people clone a large dataset. I am working on a separate project (using the template to start) that will demonstrate the value of PMTiles better. Thanks for the suggestions!

**Reply by PhilipMathieu** [reply]
And also, props for getting a good Observable Framework demo up so fast haha

**Reply by bdon** [reply]
We're not 100% set on the Source Cooperative URL being long-term available, if your use case is limited to an area you can host a smaller basemap with `pmtiles extract` or use api.protomaps.com with a common key (free for non-commercial use)
